AutoResxTranslator/MSTranslateService.cs

```diff
@@ -0,0 +1,121 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Net.Http;
+
+namespace AutoResxTranslator
+{
+    /// <summary>
+    /// Translation service using Microsoft Cogntive service.
+    /// 
+    /// ref: https://azure.microsoft.com/en-in/services/cognitive-services/translator-text-api/
+    /// </summary>
+    public class MSTranslateService
+    {
+        private static string host = "https://api-apc.cognitive.microsofttranslator.com";
+        private static string subscriptionKey = "<YOUR-SUBSCRIPTION-KEY-HERE>";
+        private static string region = "centralindia";
+
+        public static async System.Threading.Tasks.Task<string> TranslateAsync(
+            string text,
+            string fromLanguage,
+            string toLanguage)
+        {
+            if (fromLanguage.Equals("auto") || fromLanguage.Equals(""))
+            {
+                fromLanguage = null;
+            }
+
+            string route = "/translate?api-version=3.0&to=" + toLanguage + "&from=" + fromLanguage;
+
+            try
+            {
+                object[] body = new object[] { new { Text = text } };
+                var requestBody = JsonConvert.SerializeObject(body);
+
+                using (var client = new HttpClient())
+                using (var request = new HttpRequestMessage())
+                {
+                    // Build the request.
+                    request.Method = HttpMethod.Post;
+                    request.RequestUri = new Uri(host + route);
+                    request.Content = new StringContent(requestBody, Encoding.UTF8, "application/json");
+                    request.Headers.Add("Ocp-Apim-Subscription-Key", subscriptionKey);
+                    request.Headers.Add("Ocp-Apim-Subscription-Region", region);
+
+                    // Send the request and get response.
+                    HttpResponseMessage response = await client.SendAsync(request).ConfigureAwait(false);
+
+                    if (response.StatusCode == System.Net.HttpStatusCode.OK)
+                    {
+                        // Read response as a string.
+                        string resultFromMS = await response.Content.ReadAsStringAsync();
+                        TranslationResult[] deserializedOutput = JsonConvert.DeserializeObject<TranslationResult[]>(resultFromMS);
+                        // Iterate over the deserialized results.
+                        foreach (TranslationResult o in deserializedOutput)
+                        {
+                            // Iterate over the results, return the first result
+                            foreach (Translation t in o.Translations)
+                            {
+                                return t.Text;
+                            }
+                        }
+                    }
+                    else
+                    {
+                        return "Translation failed! Exception: " + response.ReasonPhrase;
```

Comment:
Unknown status of the translation. The result should consist of a success flag and the value